### PR TITLE
Node Sass!

### PR DESCRIPTION
**Pull request by phamann**
The day has finally come! Last night node-sass 2.0 was released which finally has support for all of the hipster features we use in our sass.

This reduces our Sass compile time by ~24secs.

 Ruby Sass:
iterm

Node Sass:
iterm

node-sass

A pull-request that would make @kaelig proud.
- [ ]  browser default fonts displayed instead of guardian fonts, for all devices
- [ ] X-device testing

### CODE REVIEW

**Comment by phamann** [comment]
FYI: i'm going to do some cross-browser testing and diff the two outputs to make sure it doesn't break anything before merging.

**Comment by Calanthe** [comment]
This is great!

**Comment by sndrs** [comment]
i did a lot of testing with the pre-releases trying to get it through. i think they've covered everything in this (like nested media queries), but i found the easiest way to compare was to re-compile the sass css output with node-sass, as if it were scss. that way you get a much more similar output.

this is great though!

**Comment by phamann** [comment]
@sndrs yes I did exactly that (i.e. compiled the output directly with the node-sass cli) to get the full error warnings, which surprisingly was only one: [011eef6](https://github.com/guardian/frontend/commit/011eef61262fcb2a0fca412134c7c3c204912653)

But I still want to diff outputs and test to be 100% sure.

**Comment by kaelig** [comment]
Well done guys! We're waiting for 3.2 at the FT, but most of our modules already are compatible with 3.1 :) Can't wait!

**Comment by sndrs** [comment]
@phamann not sure if i've understood correctly, sorry if i'm just saying the same thing: meant i took the css from the sass compiler and ran it though node-sass, so you have: 
- `scss > node-sass > css`
- `scss > ruby-sass > css > node-sass > css`

is that what you meant too? they out slightly different stuff, made it easier to spot the diffs, that's all.

**Comment by kaelig** [comment]
You could use the method Etsy used to roll LibSass out: serve the LibSass compiled version only to a subset of people, and increase that portion gradually up to 100%.

**Comment by sndrs** [comment]
i guess copying their method of diffing an AST is the safest way?

**Comment by kaelig** [comment]
That too! But if you can fix stuff quickly enough that's probably not necessary. There's a lot of visual regressions that make their way to production anyway, what's a few more :p

**Comment by phamann** [comment]
Their problem was slightly different. They were moving form vanilla CSS to Sass, so a much greater chance of difference in compiled output. Whereas we are just moving form one version of Sass to another which both follow the same spec, and therefore in theory a much more similar output.

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "grunt-contrib-copy": "0.6.0",
     "grunt-contrib-jshint": "0.11.0",
     "grunt-contrib-requirejs": "0.4.4",
-    "grunt-contrib-sass": "0.8.1",
+    "grunt-sass": "git://github.com/sindresorhus/grunt-sass.git#master",
```
Comment (piuccio) [inline package.json]: :scream:

Comment (phamann) [inline package.json]: Only temporary until they release their proper tagged version, should be out later today.

**Comment by sndrs** [comment]
weirdly, i can't push to this branch because of a githook error

```
    libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: random_device failed to open /dev/urandom
.git/hooks/pre-push: line 12:  6564 Abort trap: 6           grunt validate:css validate:sass
```
will look at it later, but leaving this here incase someone knows immediately what it is

**Comment by phamann** [comment]
It's most likely because sass-lint is still using ruby sass and having some conflicts. You can by pass the git hooks by running `--no-verify` on your `git push` command.

**Comment by robertberry** [comment]
THE DAY HAS FINALLY COME

libsass

**Comment by sndrs** [comment]
hjjqt

**Comment by robertberry** [comment]
I can just picture you at home all alone guffawing because of that doggamn gif

**Comment by sndrs** [comment]
it might be my first gif on a pr

**Comment by phamann** [comment]
@sndrs your changes have appeared in the diff now. Do you still have more to add?

**Comment by sndrs** [comment]
no more no, but haven't checked the output

**Comment by phamann** [comment]
Right i've diff'ed the two outputs and am happy that it's safe to merge. Hoping @michaelwmcnamara could have a poke around on some different browsers.

**Comment by sammorrisdesign** [comment]
dog

IS THIS ACTUALLY HAPPENING?

**Comment by robertberry** [comment]
CORGI :heart:

**Comment by michaelwmcnamara** [comment]
Handy tip from Alex:  npm prune followed by npm install required to get grunt to compile happily for this branch.

**Comment by michaelwmcnamara** [comment]
Body text is displaying in times font rather than guardian fonts.

image

**Comment by gklopper** [comment]
>THE DAY HAS FINALLY COME
That was 12 days ago... SHIP or SHUT

**Comment by michaelwmcnamara** [comment]
font display issue still there - definitely wouldn't ship it yet.

**Comment by phamann** [comment]
Will re-open when updated and tested again.

**Pull request closed without merging**